Stellar-transaction.x

```diff
@@ -530,12 +530,31 @@ case HOST_FUNCTION_TYPE_INSTALL_CONTRACT_CODE:
     InstallContractCodeArgs installContractCodeArgs;
 };
 
+struct AuthorizedInvocation
+{
+    Hash contractID;
+    SCSymbol functionName;
+    SCVec args;
+    AuthorizedInvocation subInvocations<>;
```

Comment:
The name `subInvocations` is intriguing because to me it implies it does not contain an authorization for the top-level invocation. Is that true? I thought that would be false. Is there a reason to not call this `authInvocations` or `authorizedInvocations`?

Reply:
No special reason really, I can rename this. The naming is totally up to debate, that's just the first thing I came up with.

Reply:
Actually after re-reading your question, it seems like there is some confusion. These indeed don't contain the top-level invocation, because the represents the tree (so the contract id/fn name/args are related to the current node and subInvocations to the children). Maybe `authorizedSubInvocations` would be a bit more explicit, though it feels a bit too verbose.

Reply:
I guess the top-level invocation is always known, because it is in the args to the `invokeHostFunction`

Reply:
In a general case args of the top-level invocation != args that need to be signed (this is especially true when multiple signers are involved).